lib/bundles/inspec-compliance/target.rb

```diff
@@ -55,6 +55,9 @@ def self.resolve(target) # rubocop:disable PerceivedComplexity, Metrics/Cyclomat
         end
         profile_fetch_url = Compliance::API.target_url(config, profile)
       end
+      # We need to pass the token to the fetcher
+      config['token'] = Compliance::API.get_token(config)
```

Comment:
So currently this does not save to disk so if that is a requirement this approach might need to be changed.